superduperdb/components/model.py

```diff
@@ -682,6 +694,26 @@ def encode_with_schema(self, outputs):
         outputs = encoded_outputs if encoded_outputs else outputs
         return outputs
 
+    def __call__(self, outputs: t.Optional[str] = None, **kwargs):
+        from superduperdb.components.graph import IndexableNode
+
+        parent_graph = None
+        parent_models = {}
+        for k, v in kwargs.items():
+            if parent_graph is None:
+                parent_graph = v.parent_graph
+                parent_models.update(v.parent_models)
+            elif parent_graph is not None:
+                assert v.parent_graph == parent_graph, 'Cannot include 2 parent graphs'
+            parent_graph.add_edge(v.model.identifier, self.identifier, key=k)
+            parent_models[v.model.identifier] = v
+        return IndexableNode(
+            model=self,
+            parent_graph=parent_graph,
+            parent_models=parent_models,
+            identifier=outputs,
+        )
```

Comment:
Maybe we need to store the `kwargs` which are not from the upstream model.

```
model1 = ObjectModel('model_1', object=lambda x: (x + 1, x + 4))
model2 = ObjectMod`el('model_2', object=lambda x, y, k: (x + 2) * y) + k
in_ = input_node('number')
out1 = model1(x=in_)
out2 = model2(x=out1[1], y=in_, k=10000000)
graph = out2.to_graph('my_graph')
```

K is a parameter that does not belong to the task model. Supporting this feature can greatly improve the flexibility of the graph.

WDYT?